plugins/reachability/src/main/java/org/qbicc/plugin/reachability/RTAInfo.java

```diff
@@ -109,32 +193,262 @@ public void visitLiveImplementors(LoadedTypeDefinition type, Consumer<LoadedType
+                    if (impl != null && !isInvokableMethod(impl)) {
+                        rtaLog.debugf("\tnewly reachable class: enqueued implementing method:  %s", impl);
+                        invokableMethods.add(impl);
+                        deferredInstanceMethods.remove(impl); // might not be deferred, but remove is a no-op if it isn't present
+                        ctxt.enqueue(impl);
+                    }
+                }
+            }
+        }
+    }
+
+    /*
+     * RTA Helper methods.
+     */
+
+    private void addReachableInterface(LoadedTypeDefinition type) {
+        if (isReachableInterface(type)) return;
+        interfaceHierarchy.computeIfAbsent(type, t -> ConcurrentHashMap.newKeySet());
+        for (LoadedTypeDefinition i: type.getInterfaces()) {
+            addReachableInterface(i);
+            addInterfaceEdge(type, i);
+        }
+
+        // For every instance method that is not already invokable,
+        // check to see if it has the same selector as an "overridden" invokable method.
+        outer:
+        for (MethodElement im : type.getInstanceMethods()) {
+            if (!isInvokableMethod(im)) {
+                for (LoadedTypeDefinition si : type.getInterfaces()) {
+                    MethodElement sm = si.resolveMethodElementInterface(im.getName(), im.getDescriptor());
+                    if (sm != null && isInvokableMethod(sm)) {
+                        rtaLog.debugf("\tnewly reachable interface: enqueued implementing method:  %s", im);
+                        invokableMethods.add(im);
+                        ctxt.enqueue(im);
+                        continue outer;
+                    }
+                }
+            }
+        }
+    }
+
+    private void addInterfaceEdge(LoadedTypeDefinition child, LoadedTypeDefinition parent) {
+        interfaceHierarchy.computeIfAbsent(parent, t -> ConcurrentHashMap.newKeySet()).add(child);
+    }
+
+    private void addReachableClass(LoadedTypeDefinition type) {
+        if (isReachableClass(type)) return;
+        classHierarchy.computeIfAbsent(type, t -> ConcurrentHashMap.newKeySet());
+        LoadedTypeDefinition superClass = type.getSuperClass();
+        if (superClass != null) {
+            addReachableClass(superClass);
+            classHierarchy.get(superClass).add(type);
+        }
+        for (LoadedTypeDefinition i: type.getInterfaces()) {
+            addReachableInterface(i);
+            addInterfaceEdge(type, i);
+        }
+    }
+
+    private void addInitializedType(LoadedTypeDefinition type) {
+        if (isInitializedType(type)) return;
+        if (type.isInterface()) {
+            addReachableInterface(type);
+        } else {
+            addReachableClass(type);
+        }
+        initializedTypes.add(type);
+    }
+
+    private void processInvokableInstanceMethod(final MethodElement target) {
+        LoadedTypeDefinition definingClass = target.getEnclosingType().load();
```

Comment:
I had to make this change locally:

```suggestion
        if (target.isPrivate()) {
            return;
        }
        LoadedTypeDefinition definingClass = target.getEnclosingType().load();
```

Reply:
I'm not sure if it's OK to bail at this point though or if the method itself has to be added to `invokableMethods`...

Reply:
I think we should bail out one level up (when we see the invoke);  240e2ef does that instead.   We want the propagation of what was originally an invoke of a non-private method to propagate through any private methods with the same selector it stumbles across as it walks through the class/interface hierarchy looking for overrides.